### PR TITLE
improving locale font-family using postCSS silent placeholders

### LINKED ISSUE #1094: font-family for Action Button and probably other components isn't correct for Asian languages
# 🐛 Bug Report 

The easiest way to see this is to go to https://react-spectrum.adobe.com/react-spectrum/Meter.html and inspect the `Storage space` label. Watch the font-family styles.

Go up to the html tag and change the language to Japanese, 'ja'. Note font-family style for the `Storage space` label has the font-family `adobe-clean-han-japanese` set on the body but it is overwritten by other styles.

image

## 🤔 Expected Behavior

Asian languages should always use Asian fonts.

## 😯 Current Behavior

font-family with Asian language on body:(ja) is being overridden by I think by style on .spectrum

## 🔦 Context

Adobe Admin Console supports many Asian languages and Asian customers.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.1.0
| Browser          | all
| Operating System |

## 🧢 Your Company/Team

Adobe / Admin Console

## 🕷 Tracking Issue (optional)

https://github.com/adobe/spectrum-css/issues/1062

### PR DESCRIPTION
**Pull request by ktabors**
Closes #1094

Currently a proof of concept.

Making a 'class' that defines the font-family for all the locales and extending that instead of defining a single font-family for the components that define a font-family.

ran chromatic - https://www.chromatic.com/build?appId=5f0dd5ad2b5fc10022a2e320&number=76
[x] the large form stories with arabic show more differences than I'd expect.
 * I checked the default form story with arabic and large scale in an older version and this and they line up with the difference being the textfield font change of being arabic. The shifting seems to be a chromatic issue/bug.

[x] need to figure out why menu in english is different
* I looked at the two menu stories where the dropdown shifted left in default locale and medium scale. I wasn't able to reproduce the shift in storybook. The RTL dropdown width change make sense because the front changed.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

The components to test are button and textfield. These are receiving a font-family change and should see a different font-family for locales (arabic, han, etc.).
Other components we might want to look at are accordion, checkbox, menu, and radio because they use `font-family: inherit`
Also, touching how font-family is defined, but it shouldn't be changing anything.

## 🧢 Your Project:
RSP

### CODE REVIEW

**Review by LFDanLu — COMMENTED** [review]
Looks good to me for the most part, have two comments/questions (perhaps non-blocking but figured I should bring it up)

1. Textfield text seems to be too high from the bottom w/ Japanese font (maybe with others as well)? This was a problem before, so might be a non issue, not blocking as well
image


2. I remember you asked this in the core room before, but now I kinda think the `spectrum-Menu-keyboard` font family should also get the i18n font family, looks better/more consistent to me IMO. Probably should ask the localization team about this maybe?

Before:
image


After (Japanese font applied):
image

File: packages/@adobe/spectrum-css-temp/components/textfield/index.css [hunk 1]
```diff
@@ -68,7 +65,7 @@ governing permissions and limitations under the License.
   overflow: visible;
 
   /* Change the input font styles in all browsers */
-  font-family: var(--spectrum-textfield-text-font-family-fixed);
+  @extend %i18nFontFamily;
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/textfield/index.css:68]: I'm a little worried this is introducing a color that wasn't in here before? https://github.com/adobe/react-spectrum/pull/2057/files#diff-dce4300cc4c0e3cd71ef087e8c0f96b5808f4e22dc87f7a4a2de7e153edb23deR48
but I don't see anything on chromatic, lets wait until the chromatic flakiness PR is merged and then run it again on here, we may have to rebase or open a new PR on a new branch to get the right baseline

File: packages/@adobe/spectrum-css-temp/components/button/index.css [hunk 2]
```diff
@@ -62,7 +62,8 @@ governing permissions and limitations under the License.
               box-shadow var(--spectrum-global-animation-duration-100) ease-out;
 
   text-decoration: none;
-  font-family: var(--spectrum-font-family-base);
+
+  @extend %i18nFontFamily;
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/button/index.css:66]: similar, this introduces a font-size and color, so we'll have to pay close attention to these components in particular when we run chromatic again
but it looks promising :)

**Comment by snowystinger** [comment]
To address @LFDanLu's comment about Japanese font on the keyboard, I'm not sure that's what we want, we should talk to design. Usually keyboard representation is done using a monospace font

File: packages/@adobe/spectrum-css-temp/components/textfield/index.css (hunk 1 above)
Comment (ktabors) [reply packages/@adobe/spectrum-css-temp/components/textfield/index.css:68]: I think I looked at this and in most cases there was also a font-size and color defined below that would override this whole general rule. Another chromatic check is warranted.

**Comment by ktabors** [comment]
I asked the i18n team and they seemed to indicate that we'd want to use the locale's font, but I'm double checking about the monospace aspect of this. Are you suggesting we should also check with design?

**Comment by ktabors** [comment]
Do we care about https://www.chromatic.com/test?appId=5f0dd5ad2b5fc10022a2e320&id=61a6d1b6cd3539003acac365? It's a half pixel change in the label that is causing a slight change to how the SearchWithin renders with a side label in a Form.

**Comment by ktabors** [comment]
> Do we care about https://www.chromatic.com/test?appId=5f0dd5ad2b5fc10022a2e320&id=61a6d1b6cd3539003acac365? It's a half pixel change in the label that is causing a slight change to how the SearchWithin renders with a side label in a Form.

This wasn't an issue in Safari or Firefox. It happens for both large and medium scale on OSX Chrome. It looks like a half pixel rounding error on the label height, which wasn't part of this font family change. The font family change happened to the input, but no height change is observer there. The two half pixel rounding errors equal a pixel which unaligns the remaining components in the chromatic story.

**Review by snowystinger — APPROVED** [review]
lgtm in chromatic

**Review by LFDanLu — APPROVED** [review]
Chromatic looks good, confirmed that buttons and textfields have the properly font family applied:
image
image

even looks like my previous complaint about the JP text being positioned too high in the JP textfield isn't as bad? Kinda hard to tell tbh:
image

**Pull request merged**